io.typefox.yang.parent/io.typefox.yang/src/test/java/io/typefox/yang/tests/YangValidatorTest.xtend

```diff
@@ -0,0 +1,634 @@
+package io.typefox.yang.tests
+
+import io.typefox.yang.yang.Import
+import io.typefox.yang.yang.YangVersion
+import org.junit.Test
+
+import static io.typefox.yang.validation.IssueCodes.*
+
+/**
+ * Validation test for the YANG language.
+ * 
+ * @author akos.kitta
+ */
+class YangValidatorTest extends AbstractYangTest {
+
+	@Test
+	def void checkYangVersion() {
+		val it = load('''
+			module example-system {
+			  yang-version 1.2;
+			  namespace "urn:example:system";
+			  prefix "sys";
+			}
+		''')
+		assertError(root.subStatements.filter(YangVersion).head, INCORRECT_VERSION, 39, 3);
+	}
+
+	@Test
+	def void checkModule_Version_Missing() {
+		val it = load('''
+			module example-system {
+			  namespace "urn:example:system";
+			  prefix "sys";
+			}
+		''');
+		assertError(root, SUBSTATEMENT_CARDINALITY, 7, 14);
```

Comment:
in this case the error should be on namespace, since we expect the yang-version to come first.